File: modules/expo-ai-sdk/ios/DesignSystem/Typography.swift
import SwiftUI

/// Typography system matching the web app's design
extension Font {
    // MARK: - Display

    static let display = Font.system(size: 36, weight: .bold)
        .leading(.tight)

    // MARK: - Headings

    static let h1 = Font.system(size: 30, weight: .bold)
        .leading(.tight)

    static let h2 = Font.system(size: 24, weight: .semibold)
        .leading(.tight)

    static let h3 = Font.system(size: 20, weight: .semibold)

    static let h4 = Font.system(size: 18, weight: .semibold)

    // MARK: - Body

    static let body = Font.system(size: 16, weight: .regular)

    static let bodySmall = Font.system(size: 14, weight: .regular)

    // MARK: - UI

    static let label = Font.system(size: 14, weight: .medium)

    static let caption = Font.system(size: 12, weight: .regular)

    static let overline = Font.system(size: 11, weight: .medium)
        .tracking(0.5)

    // MARK: - Code

    static let code = Font.system(size: 14, weight: .regular, design: .monospaced)

    static let codeSmall = Font.system(size: 12, weight: .regular, design: .monospaced)
}

/// Text style modifiers
extension Text {
    func display() -> Text {
        self.font(.display)
    }

    func h1() -> Text {
        self.font(.h1)
    }

    func h2() -> Text {
        self.font(.h2)
    }

    func h3() -> Text {
        self.font(.h3)
    }

    func h4() -> Text {
        self.font(.h4)
    }

    func bodyText() -> Text {
        self.font(.body)
    }

    func bodySmall() -> Text {
        self.font(.bodySmall)
    }

    func label() -> Text {
        self.font(.label)
    }

    func caption() -> Text {
        self.font(.caption)
    }

    func overline() -> Text {
        self.font(.overline)
    }

    func code() -> Text {
        self.font(.code)
    }
}

File: src/design-system/typography.ts
/**
 * Typography System
 * Font sizes, weights, and line heights matching the web app
 */

export interface TypographyToken {
  fontSize: number;
  lineHeight: number;
  fontWeight?: '400' | '500' | '600' | '700';
  letterSpacing?: number;
}

export const typography = {
  // Display
  display: {
    fontSize: 36,
    lineHeight: 44,
    fontWeight: '700' as const,
    letterSpacing: -0.5,
  },

  // Headings
  h1: {
    fontSize: 30,
    lineHeight: 38,
    fontWeight: '700' as const,
    letterSpacing: -0.5,
  },
  h2: {
    fontSize: 24,
    lineHeight: 32,
    fontWeight: '600' as const,
    letterSpacing: -0.25,
  },
  h3: {
    fontSize: 20,
    lineHeight: 28,
    fontWeight: '600' as const,
  },
  h4: {
    fontSize: 18,
    lineHeight: 26,
    fontWeight: '600' as const,
  },

  // Body
  body: {
    fontSize: 16,
    lineHeight: 24,
    fontWeight: '400' as const,
  },
  bodySmall: {
    fontSize: 14,
    lineHeight: 20,
    fontWeight: '400' as const,
  },

  // UI
  label: {
    fontSize: 14,
    lineHeight: 20,
    fontWeight: '500' as const,
  },
  caption: {
    fontSize: 12,
    lineHeight: 16,
    fontWeight: '400' as const,
  },
  overline: {
    fontSize: 11,
    lineHeight: 16,
    fontWeight: '500' as const,
    letterSpacing: 0.5,
  },

  // Code
  code: {
    fontSize: 14,
    lineHeight: 20,
    fontWeight: '400' as const,
    fontFamily: 'Courier' as const, // monospace
  },
  codeSmall: {
    fontSize: 12,
    lineHeight: 18,
    fontWeight: '400' as const,
    fontFamily: 'Courier' as const,
  },
} as const;

export type TypographyVariant = keyof typeof typography;
